x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -359,12 +415,26 @@ public void cleanUpFeature(
                     + (stopTransformsResponse.getTaskFailures().isEmpty()
                         ? ""
                         : "task failures: " + stopTransformsResponse.getTaskFailures());
-                listener.onResponse(new ResetFeatureStateResponse.ResetFeatureStateStatus(this.getFeatureName(), errMsg));
+                unsetResetModeListener.onResponse(new ResetFeatureStateResponse.ResetFeatureStateStatus(this.getFeatureName(), errMsg));
             }
-        }, listener::onFailure);
+        }, unsetResetModeListener::onFailure);
+
+        ActionListener<AcknowledgedResponse> afterResetModeSet = ActionListener.wrap(
+            response -> {
+                StopTransformAction.Request stopTransformsRequest = new StopTransformAction.Request(
+                    Metadata.ALL,
+                    true,
+                    true,
+                    null,
+                    true,
+                    false
+                );
+                client.execute(StopTransformAction.INSTANCE, stopTransformsRequest, afterStoppingTransforms);
+            },
+            finalListener::onFailure
+        );
 
-        StopTransformAction.Request stopTransformsRequest = new StopTransformAction.Request(Metadata.ALL, true, true, null, true, false);
-        client.execute(StopTransformAction.INSTANCE, stopTransformsRequest, afterStoppingTransforms);
+        client.execute(SetResetModeAction.INSTANCE, SetResetModeActionRequest.enabled(), afterResetModeSet);
```

Comment:
what happens if something goes wrong between set and reset? E.g. node crashes between the 2. Will reset mode stay in cluster state in this case? What is the mitigation? Do I have to call reset again?

Reply:
> Will reset mode stay in cluster state in this case?

Yes

> What is the mitigation? Do I have to call reset again?

Yes